crates/anvil-polkadot/src/api_server/filters.rs

```diff
@@ -316,3 +342,124 @@ impl LogsFilter {
         true
     }
 }
+
+/// Filter for pending transactions
+///
+/// Monitors the transaction pool and returns newly pending transaction hashes
+/// when polled. Transactions that have been included in old blocks are automatically filtered out.
+///
+/// The filter maintains state of previously seen transactions to ensure each
+/// transaction is reported only once, even if it remains in the pending pool
+/// across multiple polls.
+pub struct PendingTransactionsFilter {
+    /// Set of transaction hashes already reported to the client
+    already_seen: HashSet<H256>,
+    /// Stream of new block notifications for detecting mined transactions
+    block_notifications: BroadcastStream<H256>,
+    /// Reference to the transaction pool for querying ready transactions
+    tx_pool: Arc<TransactionPoolHandle>,
+    /// Ethereum RPC client for fetching block transaction data
+    eth_rpc_client: EthRpcClient,
+}
+impl PendingTransactionsFilter {
+    pub fn new(
+        block_notifier: BroadcastStream<H256>,
+        tx_pool: Arc<TransactionPoolHandle>,
+        eth_rpc_client: EthRpcClient,
+    ) -> Self {
+        Self {
+            already_seen: tx_pool
+                .ready()
+                .filter_map(|tx| extract_tx_info(&tx.data).map(|(_, _, tx_info)| tx_info.hash))
+                .collect(),
+            block_notifications: block_notifier,
+            tx_pool,
+            eth_rpc_client,
+        }
+    }
+
+    /// Drains all new pending transaction hashes since the last poll.
+    ///
+    /// This method:
+    /// 1. Queries the current ready transaction pool
+    /// 2. Drains block notifications to identify mined transactions
+    /// 3. Returns only new transactions (not previously seen and not mined)
+    ///
+    /// The filter state is updated to remember all currently pending transactions,
+    /// ensuring they won't be reported again on subsequent polls.
+    async fn drain_transactions(&mut self) -> Vec<H256> {
+        // Get current ready transactions
+        let current_ready: HashSet<H256> = self
+            .tx_pool
+            .ready()
+            .filter_map(|tx| {
+                extract_tx_info(&tx.data).map(|(_, _, tx_info)| tx_info.hash).or_else(|| {
+                warn!(target: "node::filter", "Failed to extract transaction info from ready pool");
+                None
+            })
+            })
+            .collect();
+
+        // Get transactions that have been included in blocks already
+        let mut included_transactions = HashSet::new();
+        while let Some(result) = self.block_notifications.next().now_or_never().flatten() {
+            match result {
+                Ok(block_hash) => match self.fetch_block_transactions(&block_hash).await {
+                    Ok(tx_hashes) => included_transactions.extend(tx_hashes),
+                    Err(e) => {
+                        warn!(
+                            target: "node::filter",
```

Comment:
nit: lets create a `pub const LOG_TARGET: &str = "node::filter"` and use it here and in the rest of the places;